telegram/ext/_applicationbuilder.py

```diff
@@ -916,14 +921,33 @@ def concurrent_updates(self: BuilderType, concurrent_updates: Union[bool, int])
         .. seealso:: :attr:`telegram.ext.Application.concurrent_updates`
 
         Args:
-            concurrent_updates (:obj:`bool` | :obj:`int`): Passing :obj:`True` will allow for
-                ``256`` updates to be processed concurrently. Pass an integer to specify a
-                different number of updates that may be processed concurrently.
+            concurrent_updates (:obj:`bool` | :obj:`int` | :class:`BaseUpdateProcessor`): Passing
+            :obj:`True` will allow for ``256`` updates to be processed concurrently. Pass an
+            integer to specify a different number of updates that may be processed concurrently,
+            in that case :class:`telegram.ext.SimpleUpdateProcessor` is used to handle updates.
+            Pass an instance of :class:`telegram.ext.BaseUpdateProcessor` to use that instance
+            for handling updates concurrently.
 
         Returns:
             :class:`ApplicationBuilder`: The same builder with the updated argument.
         """
-        self._concurrent_updates = concurrent_updates
+        # Check if concurrent updates is bool and convert to integer
+        if concurrent_updates is True:
+            concurrent_updates = 256
+        elif concurrent_updates is False:
+            concurrent_updates = 0
+
+        # If `concurrent_updates` is an integer, create a `SimpleUpdateProcessor`
+        # instance with that integer value; otherwise, raise an error if the value
+        # is negative
+        if isinstance(concurrent_updates, int):
+            if concurrent_updates < 0:
+                raise ValueError("`concurrent_updates` must be a non-negative integer!")
```

Comment:
```suggestion
            if concurrent_updates < 1:
                raise ValueError("`concurrent_updates` must be a positive integer!")
```

Same in `BaseUpdateProcessor`. 0 concurrent updates would mean that we don't process updates at all :D